src/network/networkd-setlink.h

```diff
@@ -0,0 +1,19 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include <inttypes.h>
+
+typedef struct Link Link;
+typedef struct Request Request;
+
+typedef enum SetLinkMode {
```

Comment:
dunno, but maybe call this SetLinkOperation? after all this is about the various *operations* you want to execute on the device, not a "mode" this is in